luigi/contrib/hdfs/snakebite_client.py

```diff
@@ -0,0 +1,274 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+A luigi file system client that wraps around snakebite
+
+Originally written by Alan Brenner <alan@magnetic.com> github.com/alanbbr
+"""
+
+
+from luigi.contrib.hdfs import config as hdfs_config
+from luigi.contrib.hdfs import error as hdfs_error
+from luigi.contrib.hdfs import hadoopcli_clients as hdfs_hadoopcli_clients
+from luigi import six
+import luigi.contrib.target
+import logging
+import datetime
+import os
+
+logger = logging.getLogger('luigi-interface')
+
+
+class SnakebiteHdfsClient(hdfs_hadoopcli_clients.HdfsClient):
+    """
+    A hdfs client using snakebite. Since Snakebite has a python API, it'll be
+    about 100 times faster than the hadoop cli client, which does shell out to
+    a java program on each file system operation.
+    """
+
+    def __init__(self):
+        super(SnakebiteHdfsClient, self).__init__()
+        self._bite = None
+        self.pid = -1
+
+    @staticmethod
+    def list_path(path):
+        if isinstance(path, list) or isinstance(path, tuple):
+            return path
+        if isinstance(path, str) or isinstance(path, unicode):
```

Comment:
If I'm not missing something, the unicode part of the if is not Python 3 friendly.

The following code will be check for `basestring` (equivalent to `(str, unicode)`) in Python 2, and `str` in Python 3:

```py
if isinstance(path, six.string_types):
```